src/lakefs_spec/spec.py

```diff
@@ -90,6 +91,17 @@ def parse(path: str) -> tuple[str, str, str]:
     repo, ref, resource = results.groups()
     return repo, ref, resource
 
+def check_branch_exists_or_create(client, repository, new_branch_name:str, source_branch_name:str = 'main'):
```

Comment:
What do you think about a name like `ensure_branch`? Since in this way, we _ensure_ that the branch exists after this code executes, either by creating it or passing on error. 

There is the `ensurerepo` function in the tests as precedent - maybe this should be named in the same manner.

Reply:
Like it :)